src/ui/TableOfContent.tsx

```diff
@@ -1,10 +1,75 @@
 import React, { useState } from 'react';
-import { Text } from '@chakra-ui/react';
+import {
+  Link,
+  Menu,
+  MenuButton,
+  MenuItem,
+  MenuList,
+  Portal,
+  Text,
+} from '@chakra-ui/react';
 import { Icon, IconNames } from '@nypl/design-system-react-components';
 import { Navigator, ReaderState, WebpubManifest } from '../types';
-import { Link } from '@chakra-ui/react';
 import Button from './Button';
 import { HEADER_HEIGHT } from './Header';
+import useColorModeValue from './hooks/useColorModeValue';
+
+type TocItemProps = React.ComponentPropsWithoutRef<typeof MenuItem> & {
+  href: string;
+  title: string | undefined;
+  isActive: boolean;
+  onClickHandler: (evt: any) => void;
+};
+
+const TocItem = (props: TocItemProps) => {
+  const { href, title, isActive, onClickHandler, ...rest } = props;
+
+  const bgColor = useColorModeValue('ui.white', 'ui.black', 'ui.sepia');
+  const color = useColorModeValue('ui.black', 'ui.white', 'ui.black');
+  const activeHover = useColorModeValue('ui.white', 'ui.white', 'ui.sepia');
+  const borderColor = useColorModeValue(
+    'ui.gray.medium',
+    'gray.500',
+    'yellow.600'
+  );
+
+  const activeColor = 'ui.black';
+  const activeBgColor = 'ui.gray.light-cool';
+
```

Comment:
It might be good to separate this component and made it into a new file?

Reply:
I don't think its necessary unless the file gets too long to navigate nicely. Seems fine to me, especially because it's only used in this file